lib/open_api_spex/cast/number.ex

```diff
@@ -0,0 +1,66 @@
+defmodule OpenApiSpex.Cast.Number do
+  @moduledoc false
+  alias OpenApiSpex.Cast
+
+  @spec cast(ctx :: Cast.t()) :: {:ok, Cast.t()} | Cast.Error.t()
+  def cast(%{value: value} = ctx) when is_number(value) do
+    case cast_number(ctx) do
+      {:cast, ctx} -> cast(ctx)
+      result -> result
+    end
+  end
+
+  # TODO We need a way to distinguish numbers in (JSON) body vs in request parameters
+  # so we can reject strings values for properties of `type: :number`
+  def cast(%{value: value} = ctx) when is_binary(value) do
+    case Float.parse(value) do
+      {value, ""} -> cast(%{ctx | value: value})
+      _ -> Cast.error(ctx, {:invalid_type, :number})
+    end
+  end
+
+  def cast(ctx) do
+    Cast.error(ctx, {:invalid_type, :number})
+  end
+
+  ## Private functions
+
+  defp cast_number(%{value: value, schema: %{minimum: minimum, exclusiveMinimum: true}} = ctx)
+       when is_number(value) and is_number(minimum) do
```

Comment:
> `when is_number(value)`

Wouldn't `value` always be a number for this function?

Reply:
I think that's true, however, the corresponding function in `integer.ex` also has this guard (maybe it was left there for historical reasons) and anyway I figured, hey, "guards can't hurt, right (esp. since Elixir doesn't quite enforce type checking at compile time)?"

I can take it out if we think it's unnecessary.

Reply:
It's a minor point. No worries.